### ISSUE TITLE
URI Parameters Operation Filter behavior difference in 2.6 vs 2.5

### ISSUE THREAD

**Issue by gshackles** [openrasta/openrasta-core#185]
I still haven't completely tracked down what might have changed, but wanted to start leaving some notes here in the event someone else has any insight on things. As part of testing the 2.6 preview, I noticed that for one of my handlers that has a bunch of operations in it, one particular endpoint started throwing an error in 2.6 complaining that it found multiple matching operations, whereas in 2.5 it correctly selected the expected operation without issue.

In the handler, the two methods being returned from `UriParametersFilter` look like:

```
public OperationResult Get(int id)

public List<metadataitem> GetMetadata(Vendor vendor)
```

In the request being made there is only one input parameter, named `id`, and the first method above is the expected match. In tracing the operation, it looks as though the second is matching because `Vendor` has a `int Id { get; private set; }` property on it that it finds and seems to set successfully via reflection.

**Comment by gshackles** [comment]
While there's definitely a change in behavior here, I think I was wrong about suspecting the parameters filter. I ran the app with both 2.5 and 2.6 and captured the diagnostic logs, and can see that filter also returning both methods in 2.5. I'm assuming a later filter eliminated the second method in 2.5 whereas it's not in 2.6. Will continue to dig as I can, but for now I figured I'd also include some of the relevant logs here as well.

2.5:

```
openrasta Verbose: 0 : Incoming host request for http://ordering.ololocal.net:8081/restaurants/850
openrasta Verbose: 0 : Adding communication context data
openrasta Start: 1 : Entering OpenRastaRewriterHandler: Rewriting to original path
openrasta Start: 1 : Entering OpenRastaIntegratedHandler: Request for http://ordering.ololocal.net:8081/restaurants/850
openrasta Verbose: 0 : Incoming host request for http://ordering.ololocal.net:8081/restaurants/850
openrasta Verbose: 0 : Adding communication context data
openrasta Verbose: 0 : Found 10 operation(s) with a matching name.
openrasta Verbose: 0 : Found 0 operation(s) with matching [HttpOperation] attribute.
openrasta Information: 0 : Operation RestaurantHandler::Get(Int32 id) selected with 1 required members and 0 optional members, without request codec
openrasta Information: 0 : Operation RestaurantHandler::GetMetadata(Vendor vendor) selected with 1 required members and 0 optional members, without request codec
openrasta Verbose: 0 : Ignoring constructor, following dependencies didn't have a registration:OpenRasta.OperationModel.Interceptors.IOperationInterceptor[]
openrasta Verbose: 0 : Ignoring constructor, following dependencies didn't have a registration:OpenRasta.OperationModel.Interceptors.IOperationInterceptor[]
openrasta Information: 0 : Executing OperationResult OperationResult: type=OK, statusCode=200.
openrasta Information: 0 : Selected codec JsonDataContractCodec out of 2 codecs for entity of type restaurant and negotiated media type application/json; q=0.5.
openrasta Verbose: 0 : Codec JsonDataContractCodec selected.
openrasta Start: 1 : Entering ResponseEntityWriterContributor: Generating response entity.
openrasta Verbose: 0 : Setting Content-Length to 2603
openrasta Stop: 1 : Exiting ResponseEntityWriterContributor
openrasta Verbose: 0 : Writing http headers.
openrasta Verbose: 0 : Ignoring constructor, following dependencies didn't have a registration:OpenRasta.OperationModel.Interceptors.IOperationInterceptor[]
openrasta Stop: 1 : Exiting OpenRastaIntegratedHandler
openrasta Stop: 1 : Exiting OpenRastaRewriterHandler
openrasta Verbose: 0 : Request finished.
```

2.6:

```
openrasta Verbose: 0 : Incoming host request for http://ordering.ololocal.net:8081/restaurants/850
openrasta Verbose: 0 : Adding communication context data
openrasta Start: 1 : LoggingMiddleware (Executing middleware CleanupMiddleware)
openrasta Start: 1 : LoggingMiddleware (Executing middleware CatastrophicFailureMiddleware)
openrasta Start: 1 : LoggingMiddleware (Executing contributor CorsHandler.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (PreExecuteMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor BootstrapperContributor.IdentitySingleTap (PreExecuteMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor NullAuthenticationContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (PreExecuteMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor UriDecoratorsContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (PreExecuteMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor ResourceTypeResolverContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (RequestMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing middleware YieldBeforeNextMiddleware)
openrasta Start: 1 : LoggingMiddleware (Executing contributor ApiKeyAuthorizer.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (RequestMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor HttpMethodOverriderContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (RequestMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor HandlerResolverContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (RequestMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor OperationCreatorContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (RequestMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor OperationFilterContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (RequestMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor RequestCodecSelector.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (RequestMiddleware middleware))
openrasta Information: 0 : Operation RestaurantHandler::Get(Int32 id) selected with 1 required members and 0 optional members, without request codec
openrasta Information: 0 : Operation RestaurantHandler::GetMetadata(Vendor vendor) selected with 1 required members and 0 optional members, without request codec
openrasta Start: 1 : LoggingMiddleware (Executing contributor RequestDecoderContributor.ReadRequestEntityBody (RequestMiddleware middleware))
openrasta Error: 0 : An error of type OpenRasta.OperationModel.AmbiguousRequestException has been thrown
openrasta Error: 0 : OpenRasta.OperationModel.AmbiguousRequestException: The request has multiple operations it could match.
openrasta Error: 0 :  - Get
openrasta Error: 0 :  - GetMetadata
openrasta Error: 0 :    at OpenRasta.OperationModel.Hydrators.RequestEntityReaderHydrator.VerifySingleMatch(IEnumerable`1 operations)
openrasta Error: 0 :    at OpenRasta.OperationModel.Hydrators.RequestEntityReaderHydrator.SelectReady(IEnumerable`1 operations)
openrasta Error: 0 :    at OpenRasta.OperationModel.Hydrators.RequestEntityReaderHydrator.Read(IEnumerable`1 operations)
openrasta Error: 0 :    at OpenRasta.Pipeline.Contributors.RequestDecoderContributor.<>c__DisplayClass1_0.<.ctor>b__0(IEnumerable`1 ops)
openrasta Error: 0 :    at OpenRasta.Pipeline.Contributors.RequestDecoderContributor.<ReadRequestEntityBody>d__3.MoveNext()
openrasta Error: 0 : --- End of stack trace from previous location where exception was thrown ---
openrasta Error: 0 :    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
openrasta Error: 0 :    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
openrasta Error: 0 :    at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
openrasta Error: 0 :    at OpenRasta.Pipeline.RequestMiddleware.<Invoke>d__2.MoveNext()
openrasta Start: 1 : LoggingMiddleware (Executing contributor MultiChannelAuthorizer.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (RequestMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor AttributeChecker.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (RequestMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor OperationInvokerContributor.ExecuteOperations (RequestMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor ClientIdHeaderAppender.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (RequestMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor ApiActivityLogger.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (RequestMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing middleware ResponseRetryMiddleware)
openrasta Start: 1 : LoggingMiddleware (Executing contributor OperationResultInvokerContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (ResponseMiddleware middleware))
openrasta Information: 0 : Executing OperationResult OperationResult: type=InternalServerError, statusCode=500 System.Collections.Generic.List`1[OpenRasta.Error].
openrasta Start: 1 : LoggingMiddleware (Executing contributor ErrorPipelineContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (ResponseMiddleware middleware))
openrasta Stop: 1 : Exiting LoggingMiddleware: LoggingMiddleware (Executing contributor ErrorPipelineContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (ResponseMiddleware middleware))
openrasta Stop: 1 : Exiting LoggingMiddleware: LoggingMiddleware (Executing contributor OperationResultInvokerContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (ResponseMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor OperationResultInvokerContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (ResponseMiddleware middleware))
openrasta Information: 0 : Executing OperationResult OperationResult: type=InternalServerError, statusCode=500 .
openrasta Start: 1 : LoggingMiddleware (Executing contributor ErrorPipelineContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (ResponseMiddleware middleware))
openrasta Start: 1 : LoggingMiddleware (Executing contributor ResponseEntityCodecResolverContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (ResponseMiddleware middleware))
openrasta Information: 0 : No response codec was searched for. The response entity is null or a response codec is already set.
openrasta Start: 1 : LoggingMiddleware (Executing contributor ResponseEntityWriterContributor.WriteResponseEntity (ResponseMiddleware middleware))
openrasta Verbose: 0 : There was no response entity, not rendering.
openrasta Verbose: 0 : Writing http headers.
openrasta Verbose: 0 : Writing http header Content-Length:0
openrasta Start: 1 : LoggingMiddleware (Executing contributor EndContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (ResponseMiddleware middleware))
openrasta Stop: 1 : Exiting LoggingMiddleware: LoggingMiddleware (Executing contributor EndContributor.<OpenRasta.Pipeline.IPipeline.Notify>b__0 (ResponseMiddleware middleware))
```

**Comment by serialseb** [comment]
Object binder on properties that are readonly, check what testing code we still have around those code paths

**Comment by serialseb** [comment]
Fixed in 31660e6ec0fa4b17b1d8bfcb9ace4c053438e3ff
